
#Importing Libraries

In [ ]:


!pip install statsmodels
     


Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:

import pandas as pd
import numpy as np
from google.colab import drive
drive.mount('/content/drive/') 

Mounted at /content/drive/


In [ ]:
from statsmodels.stats.weightstats import ztest
import scipy.stats as stats

#Reading Dataset

In [1]:
data = pd.read_csv('/content/drive/MyDrive/sukma/ruler_sample_size.csv')
data.head()

NameError: ignored


#Sampling


##Simple Random Sampling



In [ ]:

sample_random = data.sample(n=100)
sample_random.shape

(100, 2)

In [ ]:
sample_random.head()

Unnamed: 0          0
82          82  29.997232
27          27  30.004341
72          72  29.997013
12          12  29.986322
46          46  29.998760

##Systematic Sampling

In [ ]:
index = np.arange(0, len(data), 4)
sample_systematic = data.iloc[index]
sample_systematic.head()

Unnamed: 0          0
0            0  29.996612
4            4  29.994487
8            8  30.003276
12          12  29.986322
16          16  29.990712

#Hypothesis Testing

##T-test

Tuan Fin memiliki sebuah pabrik yang memproduksi alat-alat tulis. Salah satu produk unggulan pada
perusahaan Tuan Fin adalah penggaris plastic 30cm. Plastik merupakan material yang rentan
terhadap suhu tinggi, sehingga dapat memuai pada suhu tertentu. Maka dari itu, untuk menjaga
keakuratan produk unggulan tersebut, Tuan Fin menugaskan Head of Quality Assurance untuk
melakukan Quality Checking berkala.
Pada saat kuartal kedua, didapatkan data panjang 100 sampel penggaris. Tugas kalian adalah bantu
Tuan Fin untuk menentukan apakah penggaris yang dihasilkan pada kuartal kedua masih dapat
dikategorikan lulus Quality Checking (panjang penggaris masih di sekitar 30cm).

Hipotesa:

    H0 = rata-rata batas panjang penggaris 30 cm
    H1 = rata-rata batas panjang penggaris yang tidak 30 cm 

Metode pengujian hipotesis:

    Menggunakan 100 sampel
    Interval kepercayaan adalah 95% (0,95)
    Karena kita tidak mengetahui varians populasi dan menggunakan 25 sampel, kita akan menggunakan uji-t untuk menguji hipotesis ini

In [ ]:


overall_0_avg = data['0'].mean()
overall_0_std = data['0'].std()

print(f'The overall 0 average is\t\t: {round(overall_0_avg,2)}')
print(f'The overall 0 standard deviation is\t: {round(overall_0_std,2)}')
     


The overall 0 average is		: 30.0
The overall 0 standard deviation is	: 0.01


In [ ]:
# stratified sampling
data_penggaris = data[data['0'] <= 30]
     

In [ ]:
# random sampling 
sample_panjang_penggaris = data['0'].sample(n=25)

In [ ]:
# t-test
ttest, pvalue = stats.ttest_1samp(sample_panjang_penggaris, overall_0_avg, alternative='less')
print(f'ttest: {round(ttest,2)}, p-value: {round(pvalue,2)}')

# confidence interval and alpha
confidence_interval = 0.95
alpha = 1 - confidence_interval

# evaluating the hypothesis
print('\n')
if pvalue < alpha:
  print('Reject the hypothesis')
  print(f'Since p-value ({round(pvalue,2)}) < alpha ({round(alpha,2)})')
else:
  print('Accept the hypothesis')
  print(f'Since p-value ({round(pvalue,2)}) > alpha ({round(alpha,2)})')
     

ttest: 0.35, p-value: 0.63


Accept the hypothesis
Since p-value (0.63) > alpha (0.05)


#Kesimpulan

ttest: 0.35, nilai-p: 0,63


Terima hipotesis
Karena nilai-p (0,63) > alfa (0,05)

Karena p-value lebih dari alpha, kami terima hipotesis nol (H0). Oleh karena itu, kami tidak terima hipotesis alternatif (H1).

Kesimpulannya, rata-rata panjang penggaris sama dengan 30 cm  dengn demikian penggaris yang dihasilkan pada kuartal kedua masih dapat dikategorikan lulus Quality Checking 